### PR TITLE
feat: add user and privilege proto definition

### PR DESCRIPTION
**Pull request by yezizp2012**
## What's changed and what's your intention?

As title, according to #2643 

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)
Part of https://github.com/singularity-data/risingwave/issues/2193

### CODE REVIEW

**Comment by neverchanje** [comment]
Do we have a plan for what SQL we want to support specifically? There are several options in Postgres's CREATE USER command, https://www.postgresql.org/docs/current/sql-createuser.html. I think we need a priority and focus on delivering a minimized feature set. I don't know what is required in the initial stage.

**Comment by yezizp2012** [comment]
> Do we have a plan for what SQL we want to support specifically? There are several options in Postgres's CREATE USER command, https://www.postgresql.org/docs/current/sql-createuser.html. I think we need a priority and focus on delivering a minimized feature set. I don't know what is required in the initial stage.

Please visit for https://github.com/singularity-data/risingwave/issues/2643 for more details. I list all syntaxes we plan to support in current stage.

File: proto/user.proto [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+message RevokePrivilegeResponse {
+  common.Status status = 1;
+  uint64 version = 2;
+}
+
+service UserService {
+  rpc CreateUser(CreateUserRequest) returns (CreateUserResponse);
+  rpc UpdateUser(UpdateUserRequest) returns (UpdateUserResponse);
```
Comment (TennyZhuang) [inline proto/user.proto:126]: Where should UpdateUser be called?

Comment (yezizp2012) [reply proto/user.proto:126]: When calling `ALTER USER xxxxx`.

**Review by liurenjie1024 — APPROVED** [review]

**Pull request merged**